catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrMetacardClientImpl.java

```diff
@@ -393,6 +404,18 @@ private Double degreesToMeters(double distance) {
                 Distance.LinearUnit.KILOMETER).getAs(Distance.LinearUnit.METER);
     }
 
+    private Integer getSolrQueryRowsProperty() {
+        try {
+            Integer rows = Integer.valueOf(System.getProperty(SOLR_QUERY_ROWS_PROPERTY));
+            if (rows < 1 || rows >= Integer.MAX_VALUE) {
```

Comment:
❗️If we keep the `System.getProperty(SOLR_QUERY_ROWS_PROPERTY)` above, we need to check that the `rows` integer isn't `null`.